cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -163,10 +163,7 @@ def interpolate_liquid(content, vars)
         template = Liquid::Template.parse(content)
 
         # Wrap nil values in LiquidNull to not have Liquid consider them as undefined
-        vars = vars.clone
-        vars.each do |key, value|
-          vars[key] = LiquidNull.new if value.nil?
-        end
+        vars = Hash[vars.map {|key, value| [key, value.nil? ? LiquidNull.new : value]}]
```

Comment:
I guess this is still a reference unless you `value.clone`. 

That may make it possible to change the values of options in liquid so that master gets different values that were set to variables in the variable parsing.